linera-chain/src/data_types.rs

```diff
@@ -594,18 +594,25 @@ impl BlockProposal {
         )
     }
 
-    /// Checks that the public key matches the owner and that the optional certificate matches
-    /// the outcome.
+    /// Checks that the original proposal, if present, matches the new one and has a higher round.
     pub fn check_invariants(&self) -> Result<(), &'static str> {
         match (&self.original_proposal, &self.content.outcome) {
-            (None, None) | (Some(OriginalProposal::Fast { .. }), None) => {}
+            (None, None) => {}
+            (Some(OriginalProposal::Fast { .. }), None) => ensure!(
+                self.content.round > Round::Fast,
```

Comment:
This relies on the derived `PartialEq` and `Eq` for `Round` enum, right? That's probably fine but it's easy to miss when refactoring. Can you add a short unit test to `Round` where we encode the expected equalities?